src/charm.py

```diff
@@ -163,9 +159,31 @@ def __init__(self, *args):
         self.framework.observe(self.on.run_sql_action, self._on_run_sql_action)
         self.framework.observe(self.on.test_tls_action, self._on_test_tls_action)
 
-    def _on_start(self, _) -> None:
+    def are_writes_running(self) -> bool:
+        """Returns whether continuous writes script is running."""
+        try:
+            os.kill(int(self.app_peer_data[PROC_PID_KEY]))
+            return True
+        except Exception:
+            return False
+
+    def _on_start(self, event: StartEvent) -> None:
         """Only sets an Active status."""
         self.unit.status = ActiveStatus()
+        if (
+            self.model.unit.is_leader()
+            and PROC_PID_KEY in self.app_peer_data
+            and not self.are_writes_running()
+        ):
+            try:
+                writes = self._get_db_writes()
+            except Exception:
+                logger.debug("Connection to db not yet available")
+                event.defer()
+                return
```

Comment:
PGB VM will take a while to re-establish connection if the principal VM is rebooted.